lib/matplotlib/lines.py

```diff
@@ -237,13 +237,22 @@ class Line2D(Artist):
         'steps': '_draw_steps_pre',
     }
 
+    # drawStyles should now be deprecated.
     drawStyles = {}
     drawStyles.update(_drawStyles_l)
     drawStyles.update(_drawStyles_s)
     # Need a list ordered with long names first:
     drawStyleKeys = (list(six.iterkeys(_drawStyles_l)) +
                      list(six.iterkeys(_drawStyles_s)))
 
+    _drawstyle_conv = {
+        'default': lambda x, y: (x, y),
+        'steps': pts_to_prestep,
```

Comment:
Can you just add these two to `STEP_LOOKUP_MAP` and correct the spelling of `step-pre` and friends to `steps-*`?  I suspect I meant to do the refactor you are doing now as part of the work where the `fill_between` feature got added, but ran out of steam (it was one of the last things to get merged before 1.5.0 iirc).